components/codec/src/error.rs

```diff
@@ -26,4 +26,10 @@ quick_error! {
     }
 }
 
+impl From<std::io::Error> for Error {
+    fn from(_err: std::io::Error) -> Error {
+        Error::UnexpectedEOF
+    }
+}
```

Comment:
This impl isn't very future proof since it assumes any IO error as an EOF. If the code ever encounters an IO error that isn't an EOF then it will quietly become an EOF without anybody knowing. If I were writing this I'd probably assert that it's EOF, or at least log an error if it isn't.

Reply:
Also, throwing away the original error like this does is an anti-pattern in my opinion. It's usually preferable to wrap the original error and return it as the `cause` of the new error.

Reply:
@breeswish I get `Error::Io` back, there maybe some other IO related errors

Reply:
Okay